lib/actions/apiV2.js

```diff
@@ -0,0 +1,533 @@
+import {
+  createQueryAction,
+  findGeometryForTrip,
+  findNearbyAmenitiesError,
+  findNearbyAmenitiesResponse,
+  findRouteError,
+  findRouteResponse,
+  findStopError,
+  findStopResponse,
+  findStopTimesForTrip,
+  findTripError,
+  findTripResponse,
+  receivedNearbyStopsError,
+  receivedNearbyStopsResponse
+} from './api'
+import { setMapZoom } from './config'
+import { zoomToStop } from './map'
+
+/**
+ * Generic helper for crafting GraphQL queries.
+ */
+function createGraphQLQueryAction(
+  query,
+  variables,
+  responseAction,
+  errorAction,
+  options
+) {
+  const endpoint = 'index/graphql'
+  const fetchOptions = {
+    body: JSON.stringify({ query, variables }),
+    headers: { 'Content-Type': 'application/json' },
+    method: 'POST'
+  }
+  return createQueryAction(endpoint, responseAction, errorAction, {
+    ...options,
+    fetchOptions
+  })
+}
+
+const findTrip = (params) =>
+  createGraphQLQueryAction(
+    `{
+        trip(id: "${params.tripId}") {
+          id: gtfsId
+          route {
+            id: gtfsId
+            agency {
+              id: gtfsId
+              name
+              url
+              timezone
+              lang
+              phone
+              fareUrl
+            }
+            shortName
+            longName
+            type
+            url
+            color
+            textColor
+            routeBikesAllowed: bikesAllowed
+            bikesAllowed
+          }
+          serviceId
+          tripHeadsign
+          directionId
+          blockId
+          shapeId
+          wheelchairAccessible
+          bikesAllowed
+          tripBikesAllowed: bikesAllowed
+
+          stops {
+			      gtfsId: id
+			      code
+			      name
+			      lat
+			      lon			
+		      }
+
+          tripGeometry {
+            length
+            points
+          }
+        }
+      }`,
+    {},
+    findTripResponse,
+    findTripError,
+    {
+      noThrottle: true,
+      postprocess: (payload, dispatch) => {
+        // FIXME: integrate into graphql request
+        dispatch(findStopTimesForTrip({ tripId: params.tripId }))
+        dispatch(findGeometryForTrip({ tripId: params.tripId }))
+      },
+      rewritePayload: (payload) => {
+        if (!payload?.data?.trip) return {}
+
+        payload.data.trip.geometry = payload.data.trip.tripGeometry
+        return payload.data.trip
+      }
+    }
+  )
+
+export function vehicleRentalQuery(
```

Comment:
I kinda like the notation `const vehicleRentalQuery... => createGraphQLQueryAction(...)` (multiple instances).

Reply:
Refactored in d2507aa982ecca23a82fb1964c5e7a7099b81d88